### PR TITLE
Improve RunMode / dev mode

### PR DESCRIPTION
**Pull request by wxiaoguang**
1.  non-dev mode is treated as prod mode, to protect users from accidentally running in dev mode if there is a typo in this value.
2. in dev mode, do not need to really exit if there are template errors, because the template errors could be fixed by developer soon and the templates get reloaded, help:
    * https://github.com/go-gitea/gitea/issues/24845#issuecomment-1557615382
3. Fine tune the mail template loading message.

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -251,7 +251,13 @@ func loadRunModeFrom(rootCfg ConfigProvider) {
...
 		RunMode = rootSec.Key("RUN_MODE").MustString("prod")
 	}
-	IsProd = strings.EqualFold(RunMode, "prod")
+
+	// non-dev mode is treated as prod mode, to protect users from accidentally running in dev mode if there is a typo in this value.
+	RunMode = strings.ToLower(RunMode)
+	if RunMode != "dev" {
+		RunMode = "prod"
+	}
+	IsProd = RunMode != "dev"
```
Comment (silverwind) [inline modules/setting/setting.go:260]: Couldn't this just be `IsProd = !strings.EqualFold(RunMode, "dev")`?

Comment (wxiaoguang) [reply modules/setting/setting.go:260]: We need to assign a clear value to `RunMode`

----

Even if you do `IsProd = !strings.EqualFold(RunMode, "dev")`, it still need more code to update `RunMode` to a correct value.

Comment (silverwind) [reply modules/setting/setting.go:260]: Ah, so RunMode is used elsewhere, I see.

**Review by lunny — APPROVED** [review]

File: modules/setting/setting.go (hunk 1 above)
Comment (jolheiser) [inline modules/setting/setting.go:260]: I wonder if this warrants a log message if it's not within the "enum" we expect.

**Review by silverwind — APPROVED** [review]

File: modules/setting/setting.go (hunk 1 above)
Comment (wxiaoguang) [reply modules/setting/setting.go:260]: I do no think it is necessary or must.

* Each time Gitea starts, there is already a "RunMode: prod" message. If it doesn't match user's expectation, they could go to the manual and config to check.
* There was no such warning before, no user complains.
* There are already a lot of messages during startup, I doubt few people really read them if there is no error.


So , I think we do not need an extra warning message here.

**Pull request merged**